### DISCUSSION TITLE
✅  iCloud usage

### DISCUSSION THREAD

**Discussion by ap2de** [isontheline/pro.webssh.net#500] [Q&A] [answered]
I have installes webssh on an ipad as well on an iphone.
I habe enabled icloud on both devices.
Unfortunately neither the ssh server nor the sftp server is being synced from one device to the other.
What further preferences have I  got to change?
Thank you for your reply

**Comment by isontheline** [comment, marked as answer]
Have you enabled this setting **inside** WebSSH?
WebSSH > Settings > iCloud > Enable

**Reply by ap2de** [reply]
Yes, I did.

**Reply by isontheline** [reply]
Strange :/

Could you enable verbose logging in order to check further your issue?

In order to enable verbose logging you can follow the steps :
1. iOS Settings
2. WebSSH
3. Settings
4. Log Level > Verbose
5. Restart WebSSH
6. Reproduce your action
7. Find the log files inside iOS > Files > WebSSH

When verbose logging is enabled, please edit a connection and change it's name, then save the connection.
Does logs have any error related to iCloud?

**Reply by ap2de** [reply]
thank you for the support!

Log shows following:
2022/01/01 18:46:41:369 [VERBOSE] No prefered language. Use system language if available.
2022/01/01 18:46:41:374 [VERBOSE] Locale Identifier : de_DE
2022/01/01 18:46:41:374 [VERBOSE] Device language : de
2022/01/01 18:46:41:374 [VERBOSE] Selected language : de
2022/01/01 18:46:41:832 [VERBOSE] Did become active [AppDelegate]
2022/01/01 18:46:41:837 [VERBOSE] No pin code set. WebSSH won't be locked.
2022/01/01 18:46:42:388 [WARN] Will fetch from iCloud without a change token
2022/01/01 18:46:42:415 [DEBUG] Launch_APP_Count_16.3 : 2
2022/01/01 18:46:42:416 [DEBUG] Launch_APP_Count : 2
2022/01/01 18:46:50:307 [VERBOSE] Saving connection...
2022/01/01 18:46:50:307 [VERBOSE] Saving (pushAllowed == 1) object B9675DE5-2D98-4F2F-BE20-6EBC16B800E6 to local database
2022/01/01 18:46:50:357 [WARN] Unable to save record zone : <CKError 0x280ebb390: "Not Authenticated" (9/1002); "CloudKit access was denied by user settings">
2022/01/01 18:46:53:853 [DEBUG] Application will resign active [AppDelegate]
2022/01/01 18:46:55:970 [DEBUG] Application Did Enter Background

as I am not a power user of ios, I am not sure, what the warning means.
why did i deny the cloudkit access?????

**Reply by isontheline** [reply]
Thank you so much for your quick feedback 🙏

Perhaps you have disabled iCloud Drive?

**Reply by ap2de** [reply]
stupid me, you are right!
works now.

**Reply by isontheline** [reply]
You're Welcome @ap2de 

Thank you again for your quick and strong feedback.

Have a nice day ☀️
